### PR TITLE
Rust toolchain updates

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR updates the rust toolchain to 1.67.1, which is currently the latest version. It should also fix `xbbs` for the cargo problem. Tested with `exa`, `alacritty` is still broken, and I'll look into that later.

### CODE REVIEW

**Comment by Dennisbonke** [comment]
Alacritty is no longer broken at build time, but we're running into pthread issues at runtime. Something to look into later. For now, let's re-enable `alacritty` on the ci.

File: patches/rust-nix/0001-managarm-initial-port.patch [hunk 1]
```diff
@@ -604,28 +764,62 @@ index 36a3601..c57bd1b 100644
...
+diff --git a/src/sys/timerfd.rs b/src/sys/timerfd.rs
+index 18acbae..0005b24 100644
+--- a/src/sys/timerfd.rs
++++ b/src/sys/timerfd.rs
+@@ -57,7 +57,7 @@ impl FromRawFd for TimerFd {
+ libc_enum! {
+     /// The type of the clock used to mark the progress of the timer. For more
+     /// details on each kind of clock, please refer to [timerfd_create(2)](https://man7.org/linux/man-pages/man2/timerfd_create.2.html).
+-    #[repr(i32)]
++    #[repr(i64)]
```
Comment (64) [inline patches/rust-nix/0001-managarm-initial-port.patch:798]: This is not really a robust change, but something we'll have to deal with when we upstream :^)

**Review by 64 — CHANGES_REQUESTED** [review]

File: patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+ s! {
+@@ -154,6 +187,19 @@ s! {
+     pub struct sigval {
+         pub sival_ptr: *mut ::c_void,
+     }
++
++    pub struct sigevent {
++        pub sigev_value: ::sigval,
++        pub sigev_signo: ::c_int,
++        pub sigev_notify: ::c_int,
```
[the remark this answers is not in the document]
Comment (no92) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:143]: I honestly fail to see where this is wrong.

[the remark this answers is not in the document]
Comment (64) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:143]: Aren't signo and notify the wrong way around?

[the remark this answers is not in the document]
Comment (64) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:143]: And the member below is missing too

[the remark this answers is not in the document]
Comment (Dennisbonke) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:143]: Yeah they are the wrong way around. I'll swap them. How would I do that last member in rust? I'm not sure how to do that.

[the remark this answers is not in the document]
Comment (ikbenlike) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:143]:
```rust
sigev_notify_function: fn(sigval) -> *c_void
```

probably? I'm no rust guru though

[the remark this answers is not in the document]
Comment (Dennisbonke) [reply patches/rust-libc/0004-managarm-More-additions-for-a-newer-nix-version.patch:142]: @64 should be fixed, ready for review.

**Review by 64 — APPROVED** [review]

**Pull request merged**